map_conversion.py

```diff
@@ -0,0 +1,250 @@
+import argparse
+from collections import defaultdict
+import json
+from typing import Dict, Iterable, Any
+
+import numpy as np
+from xml.dom.minidom import parseString
+import xml.etree.ElementTree as ET
+
+from argoverse.utils.manhattan_search import compute_point_cloud_bbox
+from nuscenes.map_expansion.map_api import NuScenesMap
+from nuscenes.map_expansion import arcline_path_utils
+
+
+"""
+Converts the nuScenes map into the Argoverse format.
+The nuscenes map is a json file, and is converted to an xml for use with argoverse codebase.
+
+"""
+
+filename_to_id = {
+    "boston-seaport": "BSP_10318",
+    "singapore-hollandvillage": "SHV_10320",
+    "singapore-onenorth": "SON_10322",
+    "singapore-queenstown": "SQT_10324",
+}
+
+
+def populate_lane_dict(
+    nusc_map: NuScenesMap, root: ET.Element, data: Dict[str, Iterable[Any]]
+) -> Dict[str, Iterable[int]]:
+    """
+    Return a map that links a nuscenes 'way' token to all the nodes that are a part of that way.
+    Also adds all these nodes to the xml map.
+    """
+    lane_dict = defaultdict(list)
+
+    # Map x,y value of node to the new(argo) node id in the xml
+    present_nodes = {}  # k:tuple, v: node_id
+    global_id = 0  # New id for nodes in the xml
+
+    # Loop over all lanes in the json to populate the lane to node dictionary
+    for way in data["lane"]:
+        if way["token"] == "8f23d3ed-1089-4fcf-a178-a174abc69938":
```

Comment:
is this the broken lane id "8f23d3ed-1089-4fcf-a178-a174abc69938"? i think it was fixed in https://github.com/nutonomy/nuscenes-devkit#map-versions. I can download the new version to skynet if you'd like the updated one

Reply:
Yes, this is the broken lane id they had in the previous version. If you could do that, we should be able to remove these lines

Reply:
Cool, we can do in a different PR. I'll go and merge now.